File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/ui/screens/dashboard/DashboardScreen.kt
package com.gym.gymapp.ui.screens.dashboard

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.TrendingUp
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Checklist
import androidx.compose.material.icons.filled.Groups
import androidx.compose.material.icons.filled.Payments
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.material3.pulltorefresh.PullToRefreshBox
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.gym.gymapp.ui.components.AppLoader
import com.gym.gymapp.ui.components.MemberItem
import com.gym.gymapp.ui.viewmodels.DashboardViewModel
import org.koin.compose.viewmodel.koinViewModel

@Composable
fun DashboardScreen(
    onAddMember: () -> Unit = {},
    onViewMembers: () -> Unit = {},
    onViewAttendance: () -> Unit = {},
    onViewProfile: () -> Unit = {},
    onMemberClick: (String) -> Unit = {},
    viewModel: DashboardViewModel = koinViewModel()
) {
    androidx.compose.runtime.LaunchedEffect(Unit) {
        viewModel.loadDashboardData()
    }

    BoxWithConstraints(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
    ) {
        val isTablet = maxWidth > 600.dp

        if (isTablet) {
            // Tablet Layout: 2 Columns
            Row(
                modifier = Modifier.fillMaxSize().padding(16.dp),
                horizontalArrangement = Arrangement.spacedBy(24.dp)
            ) {
                // Left Column: Overview & Stats
                Column(modifier = Modifier.weight(1.2f)) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                viewModel.uiState.gymName.ifEmpty { "Gym Dashboard" },
                                color = MaterialTheme.colorScheme.onBackground,
                                fontSize = 22.sp,
                                fontWeight = FontWeight.Black
                            )
                            Text(
                                "Main Branch Overview",
                                fontSize = 14.sp,
                                color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.6f),
                                fontWeight = FontWeight.Medium
                            )
                        }
                    }

                    Spacer(Modifier.height(24.dp))
                    Text("Overview", fontWeight = FontWeight.Bold, fontSize = 16.sp)
                    Spacer(Modifier.height(16.dp))

                    Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                        StatCard("Total Members", viewModel.uiState.totalMembers, Icons.Default.Groups, MaterialTheme.colorScheme.primary, Modifier.weight(1f))
                        StatCard("Revenue", viewModel.uiState.totalRevenue, Icons.Default.Payments, MaterialTheme.colorScheme.onSurface, Modifier.weight(1f))
                    }
                    Spacer(Modifier.height(16.dp))
                    StatCard("Growth", viewModel.uiState.monthlyGrowth, Icons.AutoMirrored.Filled.TrendingUp, MaterialTheme.colorScheme.onSurface, Modifier.fillMaxWidth())
                    
                    Spacer(Modifier.height(32.dp))
                    Text("Quick Actions", fontWeight = FontWeight.Bold, fontSize = 16.sp)
                    Spacer(Modifier.height(16.dp))
                    Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                        Button(onClick = onAddMember, modifier = Modifier.weight(1f).height(56.dp), shape = RoundedCornerShape(16.dp)) {
                            Icon(Icons.Default.Add, null)
                            Spacer(Modifier.width(8.dp))
                            Text("Add Member")
                        }
                        OutlinedButton(onClick = onViewAttendance, modifier = Modifier.weight(1f).height(56.dp), shape = RoundedCornerShape(16.dp)) {
                            Icon(Icons.Default.Checklist, null)
                            Spacer(Modifier.width(8.dp))
                            Text("Attendance")
                        }
                    }
                }

                // Right Column: Recent Activity/Members
                Column(modifier = Modifier.weight(1f)) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text("Recent Members", fontWeight = FontWeight.Bold, fontSize = 16.sp)
                        Text("See More", color = MaterialTheme.colorScheme.primary, fontSize = 12.sp, fontWeight = FontWeight.Bold, modifier = Modifier.clickable { onViewMembers() })
                    }
                    Spacer(Modifier.height(16.dp))

                    PullToRefreshBox(
                        isRefreshing = viewModel.uiState.isLoading,
                        onRefresh = { viewModel.loadDashboardData() },
                        modifier = Modifier.fillMaxSize()
                    ) {
                        LazyColumn(
                            verticalArrangement = Arrangement.spacedBy(12.dp),
                            modifier = Modifier.fillMaxSize()
                        ) {
                            if (viewModel.uiState.recentMembers.isEmpty()) {
                                item { EmptyCard("No members joined yet") }
                            } else {
                                items(viewModel.uiState.recentMembers) { member ->
                                    MemberItem(
                                        name = member.name,
                                        status = member.status,
                                        isActuallyExpired = member.isExpired ?: false,
                                        joinDate = member.joiningDateDisplay ?: "N/A",
                                        expiryDate = member.expiryDateDisplay ?: "N/A",
                                        phone = member.phone,
                                        imageUrl = member.photoUrl,
                                        onClick = { onMemberClick(member.id) }
                                    )
                                }
                            }
                        }
                    }
                }
            }
        } else {
            // Mobile Layout: Single Column Scroll
            PullToRefreshBox(
                isRefreshing = viewModel.uiState.isLoading,
                onRefresh = { viewModel.loadDashboardData() }
            ) {
                LazyColumn(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(horizontal = 16.dp),
                    verticalArrangement = Arrangement.spacedBy(20.dp)
                ) {
                    item {
                        Spacer(Modifier.height(20.dp))
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column {
                                Text(
                                    viewModel.uiState.gymName.ifEmpty { "Gym Dashboard" },
                                    color = MaterialTheme.colorScheme.onBackground,
                                    fontSize = 18.sp,
                                    fontWeight = FontWeight.Black
                                )
                                Text(
                                    "Main Branch Overview",
                                    fontSize = 13.sp,
                                    color = MaterialTheme.colorScheme.onBackground.copy(alpha = 0.6f),
                                    fontWeight = FontWeight.Medium
                                )
                            }
                        }
                    }

                    if (viewModel.uiState.isLoading) {
                        item { AppLoader(isFullPage = true) }
                    } else {
                        item {
                            Text("Overview", color = MaterialTheme.colorScheme.onBackground, fontSize = 14.sp, fontWeight = FontWeight.Bold, modifier = Modifier.padding(top = 4.dp))
                        }

                        item {
                            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                                ) {
                                    StatCard("Total Members", viewModel.uiState.totalMembers, Icons.Default.Groups, MaterialTheme.colorScheme.primary, Modifier.weight(1f))
                                    StatCard("Revenue", viewModel.uiState.totalRevenue, Icons.Default.Payments, MaterialTheme.colorScheme.onSurface, Modifier.weight(1f))
                                }
                                StatCard("Growth", viewModel.uiState.monthlyGrowth, Icons.AutoMirrored.Filled.TrendingUp, MaterialTheme.colorScheme.onSurface, Modifier.fillMaxWidth())
                            }
                        }


                        item {
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                verticalAlignment = Alignment.CenterVertically,
                                horizontalArrangement = Arrangement.SpaceBetween
                            ) {
                                Text("Recent Members", color = MaterialTheme.colorScheme.onBackground, fontSize = 14.sp, fontWeight = FontWeight.Bold)
                                Text("See More", color = MaterialTheme.colorScheme.primary, fontSize = 11.sp, fontWeight = FontWeight.Bold, modifier = Modifier.clickable { onViewMembers() })
                            }
                        }

                        if (viewModel.uiState.recentMembers.isEmpty()) {
                            item { EmptyCard("No members joined yet") }
                        } else {
                            items(viewModel.uiState.recentMembers.take(5)) { member ->
                                MemberItem(
                                    name = member.name,
                                    status = member.status,
                                    isActuallyExpired = member.isExpired ?: false,
                                    joinDate = member.joiningDateDisplay ?: "N/A",
                                    expiryDate = member.expiryDateDisplay ?: "N/A",
                                    phone = member.phone,
                                    imageUrl = member.photoUrl,
                                    onClick = { onMemberClick(member.id) }
                                )
                            }
                        }
                    }

                    item { Spacer(modifier = Modifier.height(24.dp)) }
                }
            }
        }
    }
}


@Composable
fun EmptyCard(text: String) {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(80.dp)
            .clip(RoundedCornerShape(20.dp))
            .background(MaterialTheme.colorScheme.surface),
        contentAlignment = Alignment.Center
    ) {
        Text(text, color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f), fontSize = 13.sp)
    }
}

@Composable
fun StatCard(
    title: String,
    value: String,
    icon: ImageVector,
    accentColor: Color,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier,
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = if (accentColor == MaterialTheme.colorScheme.primary) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.surface)
    ) {
        Column(modifier = Modifier.padding(20.dp)) {
            Box(
                modifier = Modifier
                    .size(32.dp)
                    .clip(CircleShape)
                    .background(
                        if (accentColor == MaterialTheme.colorScheme.primary) MaterialTheme.colorScheme.onPrimary.copy(
                            alpha = 0.1f
                        ) else MaterialTheme.colorScheme.onSurface.copy(alpha = 0.1f)
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    icon,
                    contentDescription = null,
                    tint = if (accentColor == MaterialTheme.colorScheme.primary) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface,
                    modifier = Modifier.size(18.dp)
                )
            }
            Spacer(modifier = Modifier.height(16.dp))
            Text(
                value,
                fontSize = 20.sp,
                fontWeight = FontWeight.Black,
                color = if (accentColor == MaterialTheme.colorScheme.primary) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface
            )
            Text(
                title,
                fontSize = 10.sp,
                color = if (accentColor == MaterialTheme.colorScheme.primary) MaterialTheme.colorScheme.onPrimary.copy(
                    alpha = 0.6f
                ) else MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
                fontWeight = FontWeight.SemiBold
            )
        }
    }
}

@Composable
fun ActivityRow(memberId: String, time: String, onClick: () -> Unit) {
    Card(
        modifier = Modifier.fillMaxWidth().clickable { onClick() },
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface)
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Box(
                modifier = Modifier
                    .size(40.dp)
                    .clip(CircleShape)
                    .background(MaterialTheme.colorScheme.onSurface.copy(alpha = 0.1f)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.Default.Person,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f),
                    modifier = Modifier.size(20.dp)
                )
            }
            Spacer(modifier = Modifier.width(12.dp))
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    "Member ${memberId.take(6).uppercase()}",
                    color = MaterialTheme.colorScheme.onSurface,
                    fontWeight = FontWeight.Bold,
                    fontSize = 14.sp
                )
                Text(
                    "Check-in successful",
                    fontSize = 11.sp,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                )
            }
            Text(
                time.split("T").getOrNull(1)?.take(5) ?: "Just now",
                color = MaterialTheme.colorScheme.primary,
                fontSize = 12.sp,
                fontWeight = FontWeight.Bold
            )
        }
    }
}

File: gym_flutter_app/lib/ui/screens/dashboard_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../widgets/stat_card.dart';
import '../widgets/member_item.dart';
import '../../logic/providers/dashboard_provider.dart';
import 'member_list_screen.dart';
import 'member_profile_screen.dart';
import 'attendance_screen.dart';
import 'add_member_screen.dart';

class DashboardScreen extends ConsumerWidget {
  const DashboardScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final uiState = ref.watch(dashboardProvider);
    final theme = Theme.of(context);

    return Scaffold(
      backgroundColor: theme.colorScheme.surface,
      body: LayoutBuilder(
        builder: (context, constraints) {
          final isTablet = constraints.maxWidth > 600;

          if (isTablet) {
            return _TabletLayout(uiState: uiState);
          } else {
            return _MobileLayout(uiState: uiState, ref: ref);
          }
        },
      ),
    );
  }
}

class _MobileLayout extends StatelessWidget {
  final DashboardUIState uiState;
  final WidgetRef ref;

  const _MobileLayout({required this.uiState, required this.ref});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return RefreshIndicator(
      onRefresh: () => ref.read(dashboardProvider.notifier).loadDashboardData(),
      child: ListView(
        padding: const EdgeInsets.symmetric(horizontal: 16),
        children: [
          const SizedBox(height: 60),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      uiState.gymName.isEmpty ? "Gym Dashboard" : uiState.gymName,
                      style: theme.textTheme.headlineSmall?.copyWith(
                        fontWeight: FontWeight.w900,
                        color: theme.colorScheme.onSurface,
                      ),
                      overflow: TextOverflow.ellipsis,
                    ),
                    Text(
                      "Main Branch Overview",
                      style: theme.textTheme.bodyMedium?.copyWith(
                        color: theme.colorScheme.onSurface.withValues(alpha: 0.6),
                        fontWeight: FontWeight.w500,
                      ),
                    ),
                  ],
                ),
              ),
              const SizedBox(width: 16),
              Image.asset(
                'assets/icon.png',
                height: 42,
              ),
            ],
          ),
          const SizedBox(height: 20),
          if (uiState.isLoading)
            const Center(child: CircularProgressIndicator())
          else ...[
            Text(
              "Overview",
              style: theme.textTheme.titleSmall?.copyWith(
                fontWeight: FontWeight.bold,
                color: theme.colorScheme.onSurface,
              ),
            ),
            const SizedBox(height: 12),
            Row(
              children: [
                Expanded(
                  child: StatCard(
                    title: "Total Members",
                    value: uiState.totalMembers,
                    icon: Icons.groups,
                    accentColor: theme.colorScheme.primary,
                  ),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: StatCard(
                    title: "Revenue",
                    value: uiState.totalRevenue,
                    icon: Icons.payments,
                    accentColor: theme.colorScheme.onSurface,
                  ),
                ),
              ],
            ),
            const SizedBox(height: 12),
            StatCard(
              title: "Growth",
              value: uiState.monthlyGrowth,
              icon: Icons.trending_up,
              accentColor: theme.colorScheme.onSurface,
            ),
            const SizedBox(height: 24),
            Text(
              "Quick Actions",
              style: theme.textTheme.titleSmall?.copyWith(
                fontWeight: FontWeight.bold,
                color: theme.colorScheme.onSurface,
              ),
            ),
            const SizedBox(height: 12),
            Row(
              children: [
                Expanded(
                  child: _ActionButton(
                    label: "Add Member",
                    icon: Icons.add,
                    isPrimary: true,
                    onPressed: () {
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (context) => const AddMemberScreen()),
                      );
                    },
                  ),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: _ActionButton(
                    label: "Attendance",
                    icon: Icons.checklist,
                    isPrimary: false,
                    onPressed: () {
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (context) => const AttendanceScreen()),
                      );
                    },
                  ),
                ),
              ],
            ),
            const SizedBox(height: 24),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text(
                  "Recent Members",
                  style: theme.textTheme.titleSmall?.copyWith(
                    fontWeight: FontWeight.bold,
                    color: theme.colorScheme.onSurface,
                  ),
                ),
                TextButton(
                  onPressed: () {
                    Navigator.push(
                      context,
                      MaterialPageRoute(builder: (context) => const MemberListScreen()),
                    );
                  },
                  child: Text(
                    "See More",
                    style: TextStyle(
                      color: theme.colorScheme.primary,
                      fontSize: 11,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
              ],
            ),
            if (uiState.recentMembers.isEmpty)
              const _EmptyCard(text: "No members joined yet")
            else
              ListView.separated(
                shrinkWrap: true,
                physics: const NeverScrollableScrollPhysics(),
                itemCount: uiState.recentMembers.length,
                separatorBuilder: (context, index) => const SizedBox(height: 12),
                itemBuilder: (context, index) {
                  final member = uiState.recentMembers[index];
                  return MemberItem(
                    name: member.name,
                    status: member.status,
                    isActuallyExpired: member.isExpired,
                    joinDate: member.joiningDateDisplay,
                    expiryDate: member.expiryDateDisplay,
                    phone: member.phone,
                    imageUrl: member.photoUrl,
                    onClick: () {
                      Navigator.push(
                        context,
                        MaterialPageRoute(builder: (context) => MemberProfileScreen(memberId: member.id)),
                      );
                    },
                  );
                },
              ),
            const SizedBox(height: 24),
          ],
        ],
      ),
    );
  }
}

class _TabletLayout extends StatelessWidget {
  final DashboardUIState uiState;

  const _TabletLayout({required this.uiState});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return Padding(
      padding: const EdgeInsets.all(24.0),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Expanded(
            flex: 12,
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  uiState.gymName.isEmpty ? "Gym Dashboard" : uiState.gymName,
                  style: theme.textTheme.headlineMedium?.copyWith(
                    fontWeight: FontWeight.w900,
                  ),
                ),
                Text(
                  "Main Branch Overview",
                  style: theme.textTheme.bodyLarge?.copyWith(
                    color: theme.colorScheme.onSurface.withValues(alpha: 0.6),
                  ),
                ),
                const SizedBox(height: 32),
                const Text("Overview", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
                const SizedBox(height: 16),
                Row(
                  children: [
                    Expanded(
                      child: StatCard(
                        title: "Total Members",
                        value: uiState.totalMembers,
                        icon: Icons.groups,
                        accentColor: theme.colorScheme.primary,
                      ),
                    ),
                    const SizedBox(width: 16),
                    Expanded(
                      child: StatCard(
                        title: "Revenue",
                        value: uiState.totalRevenue,
                        icon: Icons.payments,
                        accentColor: theme.colorScheme.onSurface,
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 16),
                StatCard(
                  title: "Growth",
                  value: uiState.monthlyGrowth,
                  icon: Icons.trending_up,
                  accentColor: theme.colorScheme.onSurface,
                ),
                const SizedBox(height: 40),
                const Text("Quick Actions", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
                const SizedBox(height: 16),
                Row(
                  children: [
                    Expanded(
                      child: _ActionButton(
                        label: "Add Member",
                        icon: Icons.add,
                        isPrimary: true,
                        onPressed: () {
                          Navigator.push(
                            context,
                            MaterialPageRoute(builder: (context) => const AddMemberScreen()),
                          );
                        },
                      ),
                    ),
                    const SizedBox(width: 12),
                    Expanded(
                      child: _ActionButton(
                        label: "Attendance",
                        icon: Icons.checklist,
                        isPrimary: false,
                        onPressed: () {
                          Navigator.push(
                            context,
                            MaterialPageRoute(builder: (context) => const AttendanceScreen()),
                          );
                        },
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
          const SizedBox(width: 32),
          Expanded(
            flex: 10,
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    const Text("Recent Members", style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
                    TextButton(
                      onPressed: () {
                        Navigator.push(
                          context,
                          MaterialPageRoute(builder: (context) => const MemberListScreen()),
                        );
                      },
                      child: Text(
                        "See More",
                        style: TextStyle(
                          color: theme.colorScheme.primary,
                          fontWeight: FontWeight.bold,
                        ),
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 16),
                Expanded(
                  child: uiState.recentMembers.isEmpty
                      ? const _EmptyCard(text: "No members joined yet")
                      : ListView.separated(
                          itemCount: uiState.recentMembers.length,
                          separatorBuilder: (context, index) => const SizedBox(height: 12),
                          itemBuilder: (context, index) {
                            final member = uiState.recentMembers[index];
                            return MemberItem(
                              name: member.name,
                              status: member.status,
                              isActuallyExpired: member.isExpired,
                              joinDate: member.joiningDateDisplay,
                              expiryDate: member.expiryDateDisplay,
                              phone: member.phone,
                              imageUrl: member.photoUrl,
                              onClick: () {
                                Navigator.push(
                                  context,
                                  MaterialPageRoute(builder: (context) => MemberProfileScreen(memberId: member.id)),
                                );
                              },
                            );
                          },
                        ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

class _EmptyCard extends StatelessWidget {
  final String text;

  const _EmptyCard({required this.text});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Container(
      width: double.infinity,
      height: 80,
      decoration: BoxDecoration(
        color: theme.colorScheme.surface,
        borderRadius: BorderRadius.circular(20),
      ),
      alignment: Alignment.center,
      child: Text(
        text,
        style: TextStyle(
          color: theme.colorScheme.onSurface.withValues(alpha: 0.6),
          fontSize: 13,
        ),
      ),
    );
  }
}

class _ActionButton extends StatelessWidget {
  final String label;
  final IconData icon;
  final bool isPrimary;
  final VoidCallback onPressed;

  const _ActionButton({
    required this.label,
    required this.icon,
    required this.isPrimary,
    required this.onPressed,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return SizedBox(
      height: 56,
      child: ElevatedButton.icon(
        onPressed: onPressed,
        icon: Icon(icon),
        label: Text(label),
        style: ElevatedButton.styleFrom(
          backgroundColor: isPrimary ? theme.colorScheme.primary : Colors.transparent,
          foregroundColor: isPrimary ? Colors.black : theme.colorScheme.onSurface,
          elevation: 0,
          side: isPrimary ? BorderSide.none : BorderSide(color: theme.colorScheme.onSurface.withValues(alpha: 0.2)),
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
        ),
      ),
    );
  }
}
